torch/_subclasses/meta_utils.py

```diff
@@ -925,29 +983,57 @@ def mk_meta():
                     # Wraps a functorch tensor class (BatchedTensor, GradTrackingTensor)
                     # in a FakeTensor
                     def _to_fake_tensor(t: MetaTensorDesc):
+                        # TODO: why aren't the recursive calls going to
+                        # meta_tensor
```

Comment:
what do you mean?

Reply:
The old code for functorch fakeification did a recursive call into `_to_fake_tensor`. But this seems a bit strange to me; we ought to have recursed all the way to top level `meta_tensor`, this seems more generic and can handle more exotic compositions of wrapping in functorch.

Reply:
I didn't change it because this PR is intended to be semantics preserving

Reply:
thanks for the clarification, that sounds good. I didn't think about the recursion when we implemented this so that's why it looks like it does right now